server/utils/serviceProviderCSVGenerator.ts

```diff
@@ -0,0 +1,54 @@
+import stringify from 'csv-stringify'
+import { ServiceProviderReportReferral } from '../models/serviceProviderReportReferral'
+
+export default class ServiceProviderCSVGenerator {
+  static generate(reportReferrals: ServiceProviderReportReferral[]): string {
+    const data: string[] = []
+    const stringifier = stringify({
+      header: true,
+      columns: [
+        { key: 'referralLink', header: 'referral_link' },
+        { key: 'referralRef', header: 'referral_ref' },
+        { key: 'referralId', header: 'referral_id' },
+        { key: 'contractId', header: 'contract_id' },
+        { key: 'organisationId', header: 'organisation_id' },
+        { key: 'referringOfficerEmail', header: 'referring_officer_email' },
+        { key: 'caseworkerId', header: 'caseworker_id' },
+        { key: 'serviceUserCRN', header: 'service_user_crn' },
+        { key: 'dateReferralReceived', header: 'date_referral_received' },
+        { key: 'dateSAABooked', header: 'date_saa_booked' },
+        { key: 'dateSAAAttended', header: 'date_saa_attended' },
+        { key: 'dateFirstActionPlanSubmitted', header: 'date_first_action_plan_submitted' },
+        { key: 'dateOfFirstActionPlanApproval', header: 'date_of_first_action_plan_approval' },
+        { key: 'dateOfFirstAttendedSession', header: 'date_of_first_attended_session' },
+        { key: 'outcomesToBeAchievedCount', header: 'outcomes_to_be_achieved_count' },
+        { key: 'outcomesAchieved', header: 'outcomes_achieved' },
+        { key: 'countOfSessionsExpected', header: 'count_of_sessions_expected' },
+        { key: 'countOfSessionsAttended', header: 'count_of_sessions_attended' },
+        { key: 'endRequestedByPPAt', header: 'end_requested_by_pp_at' },
+        { key: 'endRequestedByPPReason', header: 'end_requested_by_pp_reason' },
+        { key: 'dateEOSRSubmitted', header: 'date_eosr_submitted' },
+        { key: 'concludedAt', header: 'concluded_at' },
+      ],
+    })
+
+    stringifier.on('readable', () => {
+      let row = stringifier.read()
+      while (row) {
+        data.push(row)
+        row = stringifier.read()
+      }
+    })
+
+    reportReferrals.forEach(referral => {
+      stringifier.write(referral)
+    })
+    stringifier.end()
+
+    return data.join('')
+  }
+
+  static createFilename(): string {
+    return `refer-monitor-intervention_report_${new Date().toLocaleDateString('en-CA').replace(/-/g, '')}.csv`
```

Comment:
`en-CA` is a bit confusing here, I think. You should be able to use `CalendarDay` to do this, e.g.

```suggestion
    return `refer-monitor-intervention_report_${CalendarDay.atTimeInBritain(new Date()).iso8601.replace(/-/g, '')}.csv
```

and this'll also handle the time zone comment I left above.